src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -241,54 +243,36 @@ def calculate_expression(
     if isinstance(exp, Attribute):
         return input_data[exp.name]
     if isinstance(exp, FunctionExpression):
-        kw = {}
         # evaluated_children maps to parameters passed to the function call
         evaluated_children = [
             calculate_expression(c, input_data, analyzer) for c in exp.children
         ]
-        if exp.name not in MOCK_FUNCTION_IMPLEMENTATION_MAP:
+        original_func = getattr(
+            importlib.import_module("snowflake.snowpark.functions"), exp.name
+        )
+        signatures = inspect.signature(original_func)
+        spec = inspect.getfullargspec(original_func)
+        if exp.name not in _MOCK_FUNCTION_IMPLEMENTATION_MAP:
             raise NotImplementedError(
                 f"Function {exp.name} has not been implemented yet."
```

Comment:
NIT: Let's indicate we are talking about a mock function in the error message.

Reply:
updated